xds/src/main/java/com/linecorp/armeria/xds/ListenerManager.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.xds;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.linecorp.armeria.xds.ResourceNodeType.STATIC;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.util.SafeCloseable;
+
+import io.envoyproxy.envoy.config.bootstrap.v3.Bootstrap;
+import io.envoyproxy.envoy.config.bootstrap.v3.Bootstrap.StaticResources;
+import io.envoyproxy.envoy.config.core.v3.ConfigSource;
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+import io.netty.util.concurrent.EventExecutor;
+
+final class ListenerManager implements SafeCloseable, SnapshotWatcher<ListenerSnapshot> {
+
+    private final Map<String, ListenerResourceNode> nodes = new HashMap<>();
+    private final EventExecutor eventLoop;
+    private boolean closed;
+    private final List<SnapshotWatcher<? super ListenerSnapshot>> watchers;
+
+    ListenerManager(EventExecutor eventLoop) {
+        this.eventLoop = eventLoop;
+        watchers = ImmutableList.of(this);
+    }
+
+    void initializeBootstrap(Bootstrap bootstrap, SubscriptionContext bootstrapContext) {
+        if (bootstrap.hasStaticResources()) {
+            final StaticResources staticResources = bootstrap.getStaticResources();
+            for (Listener listener: staticResources.getListenersList()) {
+                register(listener, bootstrapContext, watchers);
```

Comment:
Do we need to add the `watchers`? It seems like it doesn't do anything because `snapshotUpdated` is noop.

Reply:
This behavior is consistent with `BootstrapClusters` in that a watcher to for static resources is added so that the watcher list is never empty (so static resources are never unregistered due to closing `ListenerRoot` or `ClusterRoot`)

Added a comment which makes this clear.